warehouse/query-core/src/main/java/datawave/query/iterator/QueryIterator.java

```diff
@@ -366,6 +369,35 @@ else if (documentRange != null && (!this.isContainsIndexOnlyTerms() && this.getT
             
             this.serializedDocuments = pipelineIter;
             
+            // apply the grouping transform if requested
+            GroupingTransform groupify = getGroupingTransform();
+            if (groupify != null) {
+                
+                // transform Entries from Key,Value to Key,Document
+                Iterator<Entry<Key,Document>> docIterator = Iterators.transform(this.serializedDocuments, new Function<Entry<Key,Value>,Entry<Key,Document>>() {
```

Comment:
is there a better way to inline this to avoid serde in the same memory space?

Reply:
When group.fields is turned on,  return.fields gets set to the same grouped fields so the documents are smaller and i am only serializing/deserializing the parts that i use. 
I am still thinking about better (more efficient) ways to do this.